File: examples/ruby/DbDnsWeekly.rb
# Runnable example: DNS Database Weekly (GET /v3.2/download/dbupdate/weekly/dns)
# Parameters for dbDnsWeekly (GET /v3.2/download/dbupdate/weekly/dns):
#   - date (string, optional): yyyy-MM-dd; omit for latest
require 'date'
require 'whoisfreaks'

WhoisFreaks.configure do |config|
  config.api_key["apiKey"] = "YOUR_API_KEY"   # set once
end

api = WhoisFreaks::DatabasesDNSApi.new
result = api.db_dns_weekly((Date.today - 1).to_s)
puts result

File: examples/php/DbDnsWeekly.php
<?php
// Runnable example: DNS Database Weekly (GET /v3.2/download/dbupdate/weekly/dns)
// Parameters for dbDnsWeekly (GET /v3.2/download/dbupdate/weekly/dns):
//   - date (string, optional): yyyy-MM-dd; omit for latest
require 'vendor/autoload.php';

$config = WhoisFreaks\Configuration::getDefaultConfiguration()
    ->setApiKey("apiKey", "YOUR_API_KEY");  // set once
$api = new WhoisFreaks\Api\DatabasesDNSApi(new GuzzleHttp\Client(), $config);
$result = $api->dbDnsWeekly((new DateTime("yesterday"))->format("Y-m-d"));
print_r($result);
